wdl/transforms/draft3/src/main/scala/wdl/draft3/transforms/wdlom2wom/WdlomWomExpression.scala

```diff
@@ -0,0 +1,34 @@
+package wdl.draft3.transforms.wdlom2wom
+
+import cats.syntax.validated._
+import common.validation.ErrorOr.ErrorOr
+import wdl.draft3.transforms.wdlom2wom.ValueConsumer.ops._
+import wdl.model.draft3.elements.ExpressionElement
+import wdl.model.draft3.elements.ExpressionElement.{IdentifierLookup, PrimitiveLiteralExpressionElement}
+import wom.expression.{IoFunctionSet, WomExpression}
+import wom.types.WomType
+import wom.values.{WomFile, WomValue}
+
+final case class WdlomWomExpression(expressionElement: ExpressionElement, inputMap: Map[ConsumedValue, FoundConsumedValue]) extends WomExpression {
+  override def sourceString: String = expressionElement.toString
+
+  override def inputs: Set[String] = inputMap.values.map(_.inputName).toSet
+
+  override def evaluateValue(inputValues: Map[String, WomValue], ioFunctionSet: IoFunctionSet): ErrorOr[WomValue] = expressionElement match {
+    case PrimitiveLiteralExpressionElement(v) => v.validNel
+    case IdentifierLookup(identifier) => inputValues.get(identifier).map(_.validNel).getOrElse(s"No value supplied for variable lookup: $identifier".invalidNel)
+    case _ => ??? // TODO other expression elements
+  }
+
+  override def evaluateType(inputTypes: Map[String, WomType]): ErrorOr[WomType] = expressionElement match {
+    case PrimitiveLiteralExpressionElement(v) => v.womType.validNel
+    case id: IdentifierLookup => inputMap((id: ExpressionElement).consumedValues.head).foundType.validNel
```

Comment:
why not match to `ExpressionElement` and avoid castiness (don't know what it's called to use `:`)

Reply:
I need to be specific about the `IdentifierLookup` here because other expression elements will interact with the types of their subexpressions in different ways.

So `IdentifierLookup extends ExpressionElement` but the `@typeclass` magic doesn't add the `consumedValues` on subtypes, only on `ExpressionElement` itself. I dunno if there's a way to tell the `@typeclass` that subtypes are also OK?

Reply:
re: `@typeclass` Dunno.   AFAIK typeclasses and subtypes don't usually mix.

So it looks like your logic is correct given your constraints

Reply:
Hmm I wonder whether that fact is indicating that I should actually have the logic in some new `ValueConsumer[IdentifierLookup]` so that I can call it directly from here, or indirectly from `ValueConsumer[ExpressionElement]`?